.github/actions/publish-nuget/New-NuGetPackage.ps1

```diff
@@ -16,21 +16,70 @@
 
 param([array] $Arguments)
 
+<#
+.SYNOPSIS
+    Updates the given project file with a GetPropertyValue target to retrieve MSBuild properties in a way that
+    Directory.Build.props files also take effect, then retrieves the property.
+#>
+function Get-ProjectProperty
+{
+    param (
+        [string] $ProjectFilePath,
+        [string] $PropertyName
+    )
+
+    try
+    {
+        $projectFileContent = Get-Content $ProjectFilePath -ErrorAction Stop
+
+        $newTarget = @"
+  <Target Name="GetPropertyValue">
+    <Message Importance="High" Text="`$($PropertyName)" />
+  </Target>
+"@
+
+        # Insert the new target XML string just before the closing </Project> tag.
+        $updatedProjectFileContent = $projectFileContent -replace '</Project>', "$newTarget`r`n</Project>"
+
+        # Write the updated content back to the project file.
+        Set-Content $ProjectFilePath $updatedProjectFileContent -ErrorAction Stop
+
+        $buildOutput = dotnet msbuild $ProjectFilePath /nologo /v:minimal /p:DesignTimeBuild=true /p:BuildProjectReferences=false /t:GetPropertyValue
+
+        # Restore the file content.
+        Set-Content $ProjectFilePath $projectFileContent -ErrorAction Stop
+
+        Write-Output ([string]::IsNullOrEmpty($buildOutput) ? '' : $buildOutput.Trim())
+    }
+    catch
+    {
+        Write-Error "Failed to add the GetPropertyValue target: $($_.Exception.Message)."
+    }
+}
+
 $projects = (Test-Path *.sln) ? (dotnet sln list | Select-Object -Skip 2 | Get-Item) : (Get-ChildItem *.csproj)
 
 foreach ($project in $projects)
 {
-    $projectFile = [xml](Get-Content $project)
-    $isPackable = $projectFile.SelectSingleNode('//PropertyGroup/IsPackable').InnerText
+    Write-Output "Packing $($project.Name)..."
+
+    $isPackableProperty = Get-ProjectProperty -ProjectFilePath  $project -PropertyName 'IsPackable'
+    $isPackable = [string]::IsNullOrEmpty($isPackableProperty) -or $isPackableProperty -eq 'true'
```

Comment:
I meant instead of the whole expression. Once you check that it's not `false` you don't need to separately check if it's null or empty:
![image](https://github.com/Lombiq/GitHub-Actions/assets/4007293/920e1533-1cae-439a-b668-f704efcdf9dc)